### ISSUE TITLE
Support for Bandwidth Limiting

### ISSUE THREAD

**Issue by teamscar** [Jigsaw-Code/outline-server#198]
I think it'd be great if you had an option in the manager to limit bandwidth for each key or something similar.

**Comment by yakovmanshin** [comment]
Yes, it would be great to be able to set up daily and/or monthly quotas for each user.

**Comment by trevj** [comment]
@teamscar Thanks for the suggestion. This feature has already been requested, so I'm going to mark this as a duplicate.

**Comment by trevj** [comment]
https://github.com/Jigsaw-Code/outline-server/issues/32

**Comment by cjhenck** [comment]
@yakovmanshin @teamscar may I ask your use case? Is it to prevent abuse in terms of "taking all the bandwidth", or to prevent cost overruns? (E.g. 10GB per month per access key)?

Thanks!

**Comment by krrisztian** [comment]
@cjhenck both, and even more. To limit a key to one device, quotas, speed limiting, etc.
There are multiple tickets opened on this topic. There is a search bar for a reason.

**Comment by alexlii1971** [comment]
Well, this is a top feature request, and it would be great if you can take it as a priority.

**Comment by alexlii1971** [comment]
Well, this is a top feature request, and it would be great if you can take it as a priority.

**Comment by bemasc** [comment]
@alexlii1971 This is implemented!  Please see [these instructions](https://www.reddit.com/r/outlinevpn/comments/mbmrvp/perkey_data_limits_now_in_outline_manager/) for how to use it.

**Comment by WenzzyX** [comment]
@bemasc How about limiting the maximum connection speed per key?

**Comment by crash481** [comment]
+1 vote to feature of limit bandwidth speed on different keys groups (or for each key separately, as easier in implementation)

This will be convenient to setup speed limit for different users groups.
Like I want that some users have more possibility to consume the server's overall Internet speed. 

I need possibility to create VPN server, and share keys and don't worry that my speed is twice slower because another user that I grant key now also downloading huge files. 
I want to give key, and know that anytime he will use only 10mb/s maximum, and i will use other part of bandwidth 

(for example 90mb/s of total 100mb/s, but in total implementation will use it 50mb/s both)

I will be glad if you found it useful and implement it soon. Also with access via API setup
